webhook_spec.yaml

```diff
@@ -0,0 +1,215 @@
+      requestBody:
+        description: Delta sync payload
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/WebhookSyncRequest'
+            examples:
+              DeltaSyncAdd:
+                summary: Segment sync to add users
+                value:
+                  id_token: '7a36ae58-c4b9-11eb-8529-0242ac130003'
+                  action_type: segment.users.sync
+                  customer_config:
+                    fields:
+                    - field_id: workspace
+                      field_display_name: Workspace
+                      value:
+                        id: workspace_1
+                        display_name: Workspace 1
+                  data:
+                    segment:
+                      id: 1338065
+                      name: Free Customers
+                    sync_info:
+                      page_number: 1
+                      total_pages: 5
+                      sync_task_id: 'afe74af0-496e-11ec-81d3-0242ac130003'
+                    add:
+                      - id: user1@email.com
+                      - id: user2@email.com
+              DeltaSyncRemove:
+                summary: Segment sync to remove users
+                value:
+                  id_token: '7a36ae58-c4b9-11eb-8529-0242ac130003'
+                  action_type: segment.users.sync
+                  customer_config:
+                    fields:
+                      - field_id: workspace
+                        field_display_name: Workspace
+                        value:
+                          id: workspace_1
+                          display_name: Workspace 1
+                  data:
+                    segment:
+                      id: 1338065
+                      name: Free Customers
+                    sync_info:
+                      page_number: 1
+                      total_pages: 5
+                      sync_task_id: 'afe74af0-496e-11ec-81d3-0242ac130003'
+                    remove:
+                      - id: user3@email.com
+                      - id: user4@email.com
+        required: true
+      responses:
+        200:
+          description: OK
+        403:
+          $ref: '#/components/responses/Unauthorized'
+        429:
+          $ref: '#/components/responses/TooManyRequests'
+        default:
+          $ref: '#/components/responses/Default' 
+components:
+  schemas:
+    WebhookSyncRequest:
+      properties:
+        id_token:
+          type: string
+          description: The identifier for customer
+          format: uuid
+        action_type:
+          type: string
+          enum: [segment.users.sync]
+          description: Webhook action type
+        customer_config:
+          $ref: '#/components/schemas/CustomerConfig'
+        data:
+          $ref: '#/components/schemas/SegmentDeltaSync'
+    CustomerConfig:
+      description: Existing customer config if any (this will be empty for now)
```

Comment:
This will be empty for open beta right? if so, I want to update the example to be empty too.

Reply:
that's correct, it's not being used right now. 👍 good catch.